runtime/evaluate.cc

```diff
@@ -420,14 +418,11 @@ class evaluator : public node_visitor {
   }
 
   void visit(const make_buffer* op) override {
-    std::size_t rank = op->dims.size();
     // Allocate a buffer with space for its dims on the stack.
-    char* storage = reinterpret_cast<char*>(alloca(sizeof(raw_buffer) + sizeof(dim) * rank));
```

Comment:
TBH, I prefer the old code. I think a macro to wrap the alloca call (like you did for `copy_dims`) makes sense, but I don't think the new raw_buffer constructor + associated changes is an improvement. `make_buffer` in particular is intended to be a very lightweight operation, that I think this PR increases the cost of substantially, because it initializes all of the buffer and dims memory, and then reinitializes it all again later. 

I get that the old code is a little repetitive, but it's not the same in each case. I thought a little bit about how to make the code more concise, but didn't come up with anything I was satisfied with myself.

Reply:
I respectfully disagree, but this is not a hill I'm gonna die on. Closing.